### PR TITLE
Add benchmarking and perf improvements to Lisa.

### PR DESCRIPTION
**Pull request by pmclanahan**
This is a branch and PR for perf work for lisa.py.

@jgmize feel free to discuss this and send commits here.
I made this a branch on the moz repo for collaboration.

### CODE REVIEW

File: smithers/lisa.py [hunk 1]
```diff
@@ -159,9 +175,13 @@ def main():
         # `rate` param on the gauge to avoid getting the length
         # of the Redis list every time.
         counter += 1
-        if counter >= 1000:
-            counter = 0
-            statsd.gauge('queue.geoip', redis.llen(rkeys.IPLOGS))
+        if args.benchmark:
+            if not counter % 1000:
+                print counter
```
Comment (jgmize) [inline smithers/lisa.py]: I know you're only printing out every 1000, but any print statements at all can have a surprising effect on performance.

File: smithers/lisa.py [hunk 2]
```diff
@@ -159,9 +175,13 @@ def main():
...
+        else:
+            if counter >= 1000:
+                counter = 0
+                statsd.gauge('queue.geoip', redis.llen(rkeys.IPLOGS))
```
Comment (jgmize) [inline smithers/lisa.py:186]: This isn't included in the benchmark, but could possibly have an effect on the performance.

**Comment by jgmize** [comment]
r+wc

The redis pipelining is a huge improvement, and I don't see any issues running multiple lisa procs with this, which I believe is the most practical solution that we can implement in the time we have.  I believe we could get some significant benefits with python3+asyncio, but as discussed on irc, that unfortunately isn't a realistic option for us before launch.

**Comment by pmclanahan** [comment]
If it'd be significantly better we could possibly run only lisa on py3....

**Comment by pmclanahan** [comment]
I think we're good to go here. Anything you'd like to see changed before merging @jgmize ?

File: smithers/lisa.py [hunk 3]
```diff
@@ -159,9 +176,14 @@ def main():
...
-            counter = 0
-            statsd.gauge('queue.geoip', redis.llen(rkeys.IPLOGS))
+        if args.benchmark:
+            if not counter % 1000:
+                pipe.execute()
+        else:
+            if counter >= 1000:
+                pipe.execute()
+                counter = 0
+                statsd.gauge('queue.geoip', redis.llen(rkeys.IPLOGS))
```
Comment (jgmize) [inline smithers/lisa.py:186]: For improved benchmarking accuracy, shouldn't we include this in the args.benchmark branch?

Comment (pmclanahan) [inline smithers/lisa.py:186]: probably. I think we're done running benches for now. We can move this though.

**Comment by pmclanahan** [comment]
Thanks. Merging. We can address further things in another PR.

**Pull request merged**